models/src/anemoi/models/migrations/__init__.py

```diff
@@ -0,0 +1,691 @@
+    Any
+        An Unpickler wrapper for torch.load.
+    """
+
+    class _Unpickler(Unpickler):
+        """And Unpickler that does not fail when the pickle object has some reference to non-existing attributes.
+        This is useful to load the "migrations" key from the checkpoint regardless of import issues.
+        """
+
+        def find_class(self, module_name: str, global_name: str, /) -> Any:
+            try:
+                return super().find_class(module_name, global_name)
+            except (ImportError, AttributeError) as e:
+                attr_name = f"{module_name}.{global_name}"
+                wild_name = f"{module_name}.*"
+                if replace_attrs is False:
+                    raise e
+                if replace_attrs is True or attr_name in replace_attrs or wild_name in replace_attrs:
+                    LOGGER.debug("Missing attribute %s.%s is checkpoint. Ignoring.", module_name, global_name)
+                    return MissingAttribute
+                raise e
+
+    class UnpicklerWrapper:
+        """For torch.load's pickle_module argument.
+        A "module" with the LenientUnpickler as Unpickler.
+        """
+
+        Unpickler = _Unpickler
+
+    return UnpicklerWrapper
+
+
+def _load_ckpt(path: str | PathLike, replace_attrs: list[str] | bool = False) -> CkptType:
+    """Loads a checkpoint
+
+    Parameters
+    ----------
+    path : str | PathLike
+        Checkpoint path
+    replace_attrs : list[str] | bool, default False
+        Replace the provided attrs by a ``MissingAttribute`` object. If False, Fill not
+        try to replace attributes. If True, will replace every missing attribute. You can use
+        * as a wildcard to be replaced by any attribute in a module.
+
+    Returns
+    -------
+    CkptType
+
+    """
+    import torch
+
+    pickle_module = get_unpickler(replace_attrs)
+    ckpt = torch.load(path, map_location="cpu", pickle_module=pickle_module, weights_only=False)
+    # TODO: remove this. Only for testing
+    if _ckpt_migration_key not in ckpt:
+        ckpt[_ckpt_migration_key] = []
+    return ckpt
+
+
+class Migrator:
+    def __init__(self, migrations: Sequence[Migration] | None = None, raise_missing_migrations: bool = True) -> None:
+        """Create the migrator object
+
+        Parameters
+        ----------
+        migrations : Sequence[Migration] | None, default None
+            List of migration to execute. If None, get migrations from the current folder.
+        raise_missing_migrations : bool
+            Whether to check if there are out of order migrations missing from the checkpoint
+        """
+
+        if migrations is None:
+            migrations = _migrations_from_path(MIGRATION_PATH, __name__)
+
+        # Compatibility groups. Checkpoints cannot be migrated past their
+        # own group. This is useful to indicate when migrating checkpoints is no longer
+        # supported.
+        self._grouped_migrations: list[list[Migration]] = []
```

Comment:
How could someone pass a migrations that are not in the current folder?

Reply:
No I was just curious, i think it's better if that's removed cause I assume for debugging/understanding the flow knowing that the sync/create commands are looking at just one folder is clearer. Thanks!